lib/rule-interface/converter.rb

```diff
@@ -2,18 +2,22 @@ module RuleInterface
   module Converter
     extend self
 
-    def hash_to_drool(data_hash:, namespace:, package:)
+    def hash_to_drool(data_hash:, namespace:, package:, session:)
       payload = {}
-      payload[:lookup] = 'session'
+      payload[:lookup] = session
       payload[:commands] = []
 
+      Thread.current[:id] = 1
+
       #insert namespace object
-      payload[:commands] << insert_object(
-        data_object: {name: namespace},
-        object_type: 'namespace',
-        package: package,
-        return_object: false
-      )
+      if namespace
+        payload[:commands] << insert_object(
+          data_object: {name: namespace},
+          object_type: 'namespace',
+          package: package,
+          return_object: false
+        )
+      end
 
       data_hash.each do |object_type, data_objects|
         data_objects.each do |data_object|
```

Comment:
Should we have a check here for the case where user wants to insert single object. Something like following:
`
data_hash: {
    product: {
        id: 12,
        name: 'Blah'
    }
}
`

Here product is just a hash and not array.

I suggest to use
`data_objects = [data_objects] unless data_objects.is_a? Array`